experimental/devices/pca9548/pca9548.go

```diff
@@ -0,0 +1,155 @@
+// Copyright 2018 The Periph Authors. All rights reserved.
+// Use of this source code is governed under the Apache License, Version 2.0
+// that can be found in the LICENSE file.
+
+package pca9548
+
+import (
+	"errors"
+	"strconv"
+	"sync"
+
+	"periph.io/x/periph/conn/i2c"
+	"periph.io/x/periph/conn/i2c/i2creg"
+	"periph.io/x/periph/conn/physic"
+)
+
+// Dev is handle to a pca9548 I²C Multiplexer.
+type Dev struct {
+	c       i2c.Bus
+	address uint16
+	names   []string
+	ports   uint8
+	// mu guards port
+	mu   sync.Mutex
+	port uint8
+}
+
+// DefaultOpts is the recommended default options.
+var DefaultOpts = Opts{Address: 0x70, Ports: 8}
+
+// Opts is the pca9548 configuration.
+type Opts struct {
+	// Address pca9548 I²C Address.
+	Address uint16
+	// Ports number of physical ports on I²C Multiplexer.
+	Ports uint8
+}
+
+// Register creates a new handel to a pca9548 I²C multiplexer, and registers
+// port names with the host. These ports can then be used as any other i2c.Bus.
+// The registered port names are in the form: i2cmux/mux-ADD-I where ADD is the
+// multiplexer I²C address in hex and I is the port number.
+// example: "i2cmux-70-0" and "mux-70-0".
+func Register(bus i2c.Bus, opts *Opts) (*Dev, error) {
```

Comment:
That is a good question. And maybe its a bad way of doing it. But since the user does not interact with the mux directly except for scan() (now removed to an example) and listports() which I'm a little dubious on now. I wanted the function name to reflect what it was doing.

```golang 
	bus, err := i2creg.Open(*i2cbus)
	if err != nil {
		return fmt.Errorf("failed to open I²C: %v", err)
	}
	defer bus.Close()

	_, err := pca9548.Register(bus, &pca9548.DefaultOpts)
	if err != nil {
		t.Errorf("failed to open I²C: %v", err)
	}

	mux0, err := i2creg.Open("mux-70-0")
	if err != nil {
		return fmt.Errorf("failed to open I²C: %v", err)
	}
	defer mux0.Close()
 	....      
```

Reply:
I'd prefer to split responsibilities:
- the dev enables multiplexing, exposing multiple sub-bus
- registration of such buses
- the scan, as discussed before

There's no way for the user to unregister after.